lib/vectors.py

```diff
@@ -80,6 +80,10 @@ def __eq__(self, other_vec):
     def __str__(self):
         return str((self.x, self.y, self.z))
 
+    def distance(self, other):
```

Comment:
Since the distance is only used for comparisons, can we add the usual `distance2` function that does not apply the relatively expensive `sqrt` operation?

In `find_closest_point()`, `distance2()` would be used (no square root operation needed for comparisons).

Reply:
Distance may be useful down the line for more operations.

Reply:
I think it's fine to keep that version as well.